File: db4onqopt/core/src/com/db4o/nativequery/expr/cmp/field/StaticFieldRoot.java
/* Copyright (C) 2004 - 2006  db4objects Inc.  http://www.db4o.com */

package com.db4o.nativequery.expr.cmp.field;

import com.db4o.nativequery.expr.cmp.*;

public class StaticFieldRoot extends ComparisonOperandRoot {
	private String _className;
	
	public StaticFieldRoot(String className) {
		this._className = className;
	}

	public String className() {
		return _className;
	}

	public boolean equals(Object obj) {
		if(obj==this) {
			return true;
		}
		if(obj==null||getClass()!=obj.getClass()) {
			return false;
		}
		StaticFieldRoot casted=(StaticFieldRoot)obj;
		return _className.equals(casted._className);
	}
	
	public int hashCode() {
		return _className.hashCode();
	}
	
	public String toString() {
		return _className;
	}

	public void accept(ComparisonOperandVisitor visitor) {
		visitor.visit(this);
	}
}

File: db4on/core/src/core/com/db4o/nativequery/expr/cmp/field/StaticFieldRoot.cs
namespace com.db4o.nativequery.expr.cmp.field
{
	public class StaticFieldRoot : com.db4o.nativequery.expr.cmp.ComparisonOperandRoot
	{
		private string _className;

		public StaticFieldRoot(string className)
		{
			this._className = className;
		}

		public virtual string ClassName()
		{
			return _className;
		}

		public override bool Equals(object obj)
		{
			if (obj == this)
			{
				return true;
			}
			if (obj == null || j4o.lang.JavaSystem.GetClassForObject(this) != j4o.lang.JavaSystem.GetClassForObject
				(obj))
			{
				return false;
			}
			com.db4o.nativequery.expr.cmp.field.StaticFieldRoot casted = (com.db4o.nativequery.expr.cmp.field.StaticFieldRoot
				)obj;
			return _className.Equals(casted._className);
		}

		public override int GetHashCode()
		{
			return _className.GetHashCode();
		}

		public override string ToString()
		{
			return _className;
		}

		public override void Accept(com.db4o.nativequery.expr.cmp.ComparisonOperandVisitor
			 visitor)
		{
			visitor.Visit(this);
		}
	}
}
